docs/developer-docs/smart-contracts/write/overview.mdx

```diff
@@ -4,137 +4,332 @@ keywords: [beginner, write, resources, architecture, single canister, multi-cani
-1. The canister serves the web assets to the browser via its `http_request` endpoint that gets invoked for each HTTP request.
-1. When the JS code runs in the browser, it can call the backend canister endpoints using the [ICP JavaScript agent](/docs/current/developer-docs/web-apps/browser-js/js-request-api) library, which is analogous to `web3.js` and `ethers.js` of Ethereum.
+1. The developer writes frontend code such as HTML, JS, or CSS.
+2. The developer configures their `dfx.json` file to include a frontend canister with type "assets".
+3. The developer deploys the project. `dfx` will compile the frontend asset files into an asset canister.
+4. Users open the application in the browser by navigating to the URL `https://<canister-id>.icp0.io` or a [custom domain](/docs/current/developer-docs/web-apps/custom-domains/using-custom-domains) if one has been registered for the canister.
+5. The canister serves the web assets to the browser via its `http_request` endpoint that gets invoked for each HTTP request.
+6. When the JS code runs in the browser, it can call the backend canister endpoints using the [ICP JavaScript agent](/docs/current/developer-docs/web-apps/browser-js/js-request-api) library, which is analogous to `web3.js` and `ethers.js` of Ethereum.
 
 Since ICP supports general HTML, JS, and CSS code, any web framework should be compatible with ICP.
-SvelteKit, React, Vue have been used successfully in production.
-`dfx v0.17.0` and newer can be used to automatically generate project templates using frameworks such as these.
-[Learn more about using these frameworks](/docs/current/developer-docs/web-apps/browser-js/js-frameworks).
+For example, SvelteKit, React, and Vue have been used successfully in production. `dfx v0.17.0` and newer can be used to generate project templates that include one of these frameworks. [Learn more](/docs/current/developer-docs/web-apps/browser-js/js-frameworks).
+
+#### Limitations
+One feature of modern web frameworks that currently doesn’t work in a canister is server-side rendering (SSR). That’s because SSR requires running JS code in the canister.
+In the future, this might become possible with Azle. Until then, if SSR is required, then one solution is to host the frontend outside of ICP while keeping the core logic in the backend canister.
+
+Having no frontend at all is also a valid option for smart contracts that don’t have a UI and are callable only by users or other smart contracts.
+
+### Creating a new project
+
+<Tabs>
+<TabItem value="prereq" label="Prerequisites" default>
+
+<input type="checkbox"/> <a href="/docs/current/developer-docs/getting-started/install">Install the IC SDK.</a>
+
+</TabItem>
+</Tabs>
+
+Create a new project. When prompted, select your backend language and frontend framework of choice:
+
+```bash
+dfx new hello
```

Comment:
Not showing them is intentional. Since this page is about informing the developer of the languages and frameworks available, prompting them to create a new project and then telling them which options to choose from `dfx new` is contradictory.